packages/components/src/utils/arrayUtils.ts

```diff
@@ -0,0 +1,11 @@
+export function appendUnique<T>(
+  existing: T[],
+  incoming: T[],
+  comparedProperty: keyof T
+) {
+  const existingValues = existing.map((e) => e[comparedProperty]) ?? [];
+  return [
+    ...existing,
+    ...incoming.filter((i) => !existingValues.includes(i[comparedProperty])),
+  ];
```

Comment:
```suggestion
  const existingValues = new Set(existing.map((e) => e[comparedProperty]) ?? []);
  return [
    ...existing,
    ...incoming.filter((i) => !existingValues.has(i[comparedProperty])),
  ];
```
O(n*m) → O(m) where n=existing.length, m=incoming.length